python_modules/dagster-graphql/dagster_graphql/implementation/fetch_schedules.py

```diff
@@ -76,11 +87,17 @@ def get_schedules_for_pipeline(graphene_info, pipeline_selector):
     repository = location.get_repository(pipeline_selector.repository_name)
     external_schedules = repository.get_external_schedules()
 
-    return [
-        GrapheneSchedule(graphene_info, external_schedule=external_schedule)
-        for external_schedule in external_schedules
-        if external_schedule.pipeline_name == pipeline_selector.pipeline_name
-    ]
+    results = []
+    for external_schedule in external_schedules:
+        if external_schedule.pipeline_name != pipeline_selector.pipeline_name:
+            continue
+
+        schedule_state = graphene_info.context.instance.get_job_state(
+            external_schedule.get_external_origin_id()
+        )
+        results.append(GrapheneSchedule(external_schedule, schedule_state))
```

Comment:
this is still O(n) right? not sure if we squashed that on the frontend by calling this resolver less in an O(n) fashion

Reply:
Yeah, we fixed that in the frontend.  This is O(n) in the sense that you might have many many schedules for a particular job, which I think should not be that common?